src/main/java/org/openrewrite/java/testing/assertj/AdoptAssertJDurationAssertions.java

```diff
@@ -0,0 +1,168 @@
+    static final MethodMatcher GET_NANO_MATCHER = new MethodMatcher("java.time.Duration getNano()");
+    static final MethodMatcher GET_SECONDS_MATCHER = new MethodMatcher("java.time.Duration getSeconds()");
+    static final MethodMatcher ISEQUALTO_INT_MATCHER = new MethodMatcher("org.assertj.core.api.AbstractIntegerAssert isEqualTo(..)");
+    static final MethodMatcher ISEQUALTO_LONG_MATCHER = new MethodMatcher("org.assertj.core.api.AbstractLongAssert isEqualTo(..)");
+    static final Map<String, String> methodMap = new HashMap<String, String>() {{
+        put("getSeconds", "hasSeconds");
+        put("getNano", "hasNanos");
+        put("hasMillis", "hasSeconds");
+        put("hasSeconds", "hasMinutes");
+        put("hasMinutes", "hasHours");
+        put("hasHours", "hasDays");
+    }};
+    static List<MethodMatcher> timeUnitMatchers = Arrays.asList(
+            new MethodMatcher("org.assertj.core.api.AbstractDurationAssert hasMillis(..)", true),
+            new MethodMatcher("org.assertj.core.api.AbstractDurationAssert hasSeconds(..)", true),
+            new MethodMatcher("org.assertj.core.api.AbstractDurationAssert hasNanos(..)", true),
+            new MethodMatcher("org.assertj.core.api.AbstractDurationAssert hasMinutes(..)", true),
+            new MethodMatcher("org.assertj.core.api.AbstractDurationAssert hasHours(..)", true),
+            new MethodMatcher("org.assertj.core.api.AbstractDurationAssert hasDays(..)", true)
+    );
+
+    @Override
+    public String getDisplayName() {
+        return "Adopt AssertJ Duration assertions";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adopt AssertJ `DurationAssert` assertions for more expressive messages.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        // TODO: figure out why this Precondition check doesn't quite work
+        return Preconditions.check(Preconditions.or(
+                new UsesMethod<>("org.assertj.core.api.AbstractDurationAssert has*(..)", true),
+                new UsesMethod<>("org.assertj.core.api.AbstractLongAssert isEqualTo(..)", true),
+                new UsesMethod<>("org.assertj.core.api.AbstractIntegerAssert isEqualTo(..)", true)
+            ), new AdoptAssertJDurationAssertionsVisitor()
+        );
+    }
+
+    @SuppressWarnings("DataFlowIssue")
+    private static class AdoptAssertJDurationAssertionsVisitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation mi, ExecutionContext ctx) {
+            J.MethodInvocation m = super.visitMethodInvocation(mi, ctx);
+            if (timeUnitMatchers.stream().anyMatch(matcher -> matcher.matches(mi))) {
+                return simplifyTimeUnits(mi, ctx);
+            }else if (ISEQUALTO_INT_MATCHER.matches(mi) || ISEQUALTO_LONG_MATCHER.matches(mi)) {
```

Comment:
Formatter not yet applied here.

Reply:
Not quite sure I am clear on this, do you mean that `maybeAutoFormat` should be used here?

Reply:
I mean you've not run IntelliJ IDEA `Cmd + Alt + L`.